File: clients/csharp/Exceptions/ConnectionException.cs
namespace LimyeDB.Exceptions;

/// <summary>
/// Thrown when the client fails to connect to the LimyeDB server.
/// </summary>
public class ConnectionException : LimyeDBException
{
    public ConnectionException(string message) : base(message)
    {
    }

    public ConnectionException(string message, Exception innerException) : base(message, innerException)
    {
    }
}

File: clients/java/src/main/java/com/limyedb/exceptions/ConnectionException.java
package com.limyedb.exceptions;

/**
 * Thrown when the client fails to connect to the LimyeDB server.
 */
public class ConnectionException extends LimyeDBException {

    public ConnectionException(String message) {
        super(message);
    }

    public ConnectionException(String message, Throwable cause) {
        super(message, cause);
    }
}
